gson/src/test/java/com/google/gson/functional/SerializedNameTest.java

```diff
@@ -51,4 +53,26 @@ public void testMultipleNamesInTheSameString() {
       this.b = b;
     }
   }
+
+  public void testPatternForEnum() {
+    Gson gson = new Gson();
+    assertTrue(Pattern.compile("\\S*").matcher("REDIRECT").matches()); // verify that the pattern is correct 
+    Response target = gson.fromJson("{'error1':'AUTH_FAILURE','error2':'REDIRECT','error3':'SERVER_ERROR'}", Response.class);
+    assertEquals(ErrorCode.AUTH_FAILURE, target.error1);
+    assertEquals(ErrorCode.UNKNOWN, target.error2);
+    assertEquals(ErrorCode.SERVER_ERROR, target.error3);
+  }
+
+  private enum ErrorCode {
+    AUTH_FAILURE,
+    SERVER_ERROR,
+    /** match all non-whitespace characters. Note that pattern is applied last. */
+    @SerializedName(value="UNKNOWN", pattern="\\S*") UNKNOWN
```

Comment:
Primary use-case is to set default. I can imagine someone using a pattern to map a collection of names to a single field.

Would love to see a more user-friendly proposal.